client/src/components/JobList.tsx

```diff
@@ -3,11 +3,27 @@ import { fetchJobs, Job } from '../lib';
 import JobCard from '../components/JobCard';
 import Pagination from '../components/Pagination';
 
-export default function JobList() {
+type JobListProps = {
+  searchInput: string;
+  locationSelect: string;
+  levelSelect: string;
```

Comment:
Are the levels fixed throughout the code? If so, you might want to use a union type instead of a string:
```
levelSelect: 'Junior' | 'Midlevel' | 'Senior';
```
(Or whatever the values are.)

You would define a type for the Level where you are defining Job so you can use it everywhere.